pkgs/test/haskell/pathsModule/default.nix

```diff
@@ -0,0 +1,46 @@
+{ haskell, lib, runCommand }:
+
+let
+  ghcSets = lib.filterAttrs
+    (name: set:
+      !(lib.hasPrefix "ghc92" name) && # TODO: https://github.com/NixOS/nixpkgs/pull/200063
+      !(lib.hasPrefix "ghc94" name) && # TODO
+      !(lib.hasPrefix "ghcHEAD" name) && # TODO
+      set ? ghc && set.ghc ? bootPkgs && !set.ghc.isGhcjs or false
+    )
+    haskell.packages;
+
+  mkPkg = pkgSet: enableRelocatable:
+    haskell.lib.compose.justStaticExecutables (
+      pkgSet.callPackage (
+        { mkDerivation, ghc }:
+        mkDerivation {
+          pname = "pathsModule${lib.optionalString enableRelocatable "-reloc"}-ghc-${ghc.version}";
+          version = "0.0.0.0";
+          src = ./test-pkg;
+          inherit enableRelocatable;
+          license = lib.licenses.mit;
+          doHaddock = false;
+        }
+      ) { }
+    );
+
+  mkTest = pkgSet: relocatable:
+    let
+      testPkg = mkPkg pkgSet relocatable;
+      binPath = "${testPkg}/bin";
+    in
+      runCommand "test-bindir-output-${testPkg.name}" { } (''
+        "${binPath}/pathsModule" | grep "${binPath}"
+      '' + lib.optionalString relocatable ''
+        cp -r ${binPath}/.. reloc
+        ./reloc/bin/pathsModule | grep "$(realpath ./reloc/bin)"
+      '' + ''
+        touch $out
+      '');
```

Comment:
Its great having more tests for some of this stuff!

Would you be able to add maybe just a short english comment explaining what exactly is being tested here?  After staring at it for a while, I realized that the idea is that you can copy a package to another location, and it still returns the correct value for `getBinDir`, but it would probably be a little quicker to understand with a short comment.

Reply:
Yes, cleaning this up was the plan – however, I think we can just toss this whole PR into the garbage bin. Unfortunately it seems that Cabal requires the world (except for the core libs it seems) to be located under the same prefix for ` --enable-relocatable` to work, so it won't fly for anything in nixpkgs except trivial examples such as the one in this PR.

Reply:
Ah, that's unfortunate.  It sounded like it might be a nice feature.

Reply:
Yes, I was especially hopeful that we could (ab)use this for single output packages that use a `Paths_` module to migitate the infamous reference problem, but alas that won't fly it seems.

Reply:
I.e. I ran into this:

```
Setup: Library directory of a dependency:
"/nix/store/mii18sn6m2hd1gnxij9kyhgmk1lq8y93-Glob-0.10.2/lib/ghc-9.0.2/x86_64-linux-ghc-9.0.2/Glob-0.10.2-GyyoVNUgmMiE1e6OoNWbDt"
is not relative to the installation prefix:
"/nix/store/p7fbjkcsgg7l6p6gdqy8c0k2cq5g1jid-emanote-0.8.0.0
```